src/EnergyPlus/DataSurfaces.cc

```diff
@@ -734,4 +734,34 @@ Real64 AbsBackSide(EnergyPlusData &state, int SurfNum)
     return AbsorptanceFromExteriorBackSide + AbsorptanceFromInteriorBackSide;
 }
 
+void GetVariableAbsorptanceSurfaceList(EnergyPlusData &state)
+{
+    for (int surfNum = 1; surfNum <= state.dataSurface->TotSurfaces; ++surfNum) {
```

Comment:
This can loop over `DataSurfaces::AllHTSurfaceList`, then you don't need to check for `ConstrNum <=0`. All HT surfaces have a valid construction.

Reply:
oh nice. I'll change it to `DataSurfaces::AllHTSurfaceList`